UPGRADE.md

```diff
@@ -1,5 +1,53 @@
 # Upgrade instructions
 
+## 0.4.0
+
+* A `context` property, and methods `::setContext()` and `::getContext()` have been added to the
+  `Zenatonable` trait.
+  
+  If you have some tasks or workflows using the same names you will have to rename them.
+  
+  Before:
+  
+  ```php
+  class RegisterOrderTask implements TaskInterface
+  {
+      private $context;
+
+      public function __construct($context)
+      {
+          $this->context = $context;
+      }
+
+      public function handle()
+      {
+          // [...]
+      }
+  }
+  ```
+
+  After:
+  
+  ```php
+  class RegisterOrderTask implements TaskInterface
+  {
+      private $orderContext;
+
+      public function __construct($context)
+      {
+          $this->orderContext = $context;
+      }
+
+      public function handle()
+      {
+          // [...]
+      }
+  }
+  ```
+  
+  Please note that the `::setContext()` method is internal and must not be used. It is called by the
+  Zenaton Agent to set the runtime context of tasks and workflows.
+
```

Comment:
In node.js we have a practice to prefix method not usable per user by a "_" - don't you think it make sense? Even if unusual:  _setContext()

Reply:
I recommend using `_context` instead of `context`. It's for us an internal property, and we give to the user the possibility to use `context` as a property name for their workflows.

Reply:
This is not something we usually do in PHP. :/